libraries/chain/abi_serializer.cpp

```diff
@@ -496,10 +496,11 @@ namespace eosio { namespace chain {
       auto v_itr = variants.end();
       auto s_itr = structs.end();
 
-      auto btype = built_in_types.find(fundamental_type(rtype));
-      if( btype != built_in_types.end() ) {
+      bool var_is_array = is_szarray(rtype);
```

Comment:
I think this can be:
`bool var_is_array = is_array(rtype) || is_szarray(rtype);`
and then just `var_is_array` below.